common/src/main/java/io/netty/util/IntSupplier.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2016 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.util;
+
+/**
+ * A generic supplier for an integer value.
+ */
+public interface IntSupplier {
+    int get();
```

Comment:
> how we should handle a IOException in the supplier

We could modify the `IntSupplier` and `SelectStrategy` to allow throwing an `Exception`. Currently it doesn't look like the event loop really does anything with the exception (just logs, sleeps, and moves on). However the higher level looks like it would have more knowledge of what to do if an exception does occur.